examples/java/core/record_streaming/src/main/java/com/microsoft/bond/examples/RecordStreaming.java

```diff
@@ -0,0 +1,77 @@
+package com.microsoft.bond.examples;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Collections;
+
+import com.microsoft.bond.Deserializer;
+import com.microsoft.bond.Serializer;
+
+import com.microsoft.bond.protocol.CompactBinaryReader;
+import com.microsoft.bond.protocol.CompactBinaryWriter;
+
+import com.microsoft.bond.examples.recordstreaming.Struct;
+
+public class RecordStreaming {
+
+    private static final int NUM_OBJECTS = 9;
+
+    public static void main(final String[] args) throws IOException {
+
+        // Create a stream of serialized structs.
+        final ByteArrayOutputStream output = serializeStream();
+
+        // Read them all back.
+        final ByteArrayInputStream input = new ByteArrayInputStream(output.toByteArray());
+        deserializeStream(input);
+    }
+
+    private static ByteArrayOutputStream serializeStream() throws IOException {
+        // An output stream can be used to accumulate multiple serialized
+        // structs. This can be done by re-using the same writer over the same
+        // stream.
+        final ByteArrayOutputStream output = new ByteArrayOutputStream();
```

Comment:
`ByteArrayOutputStream` maintains a contiguous buffer. Record streaming doesn't need that. Is there a better pair of streams to use in this case?

For example, it could use a stream that just allocated buffers on demand, and then on the reading side keeps track of where and in which buffer the current position is.